codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/generator/library/shims/ServiceShim.java

```diff
@@ -125,7 +125,7 @@ private MethodSpec serviceConstructor(BuilderSpecs builderSpecs) {
         method.addStatement(dafnyDeclareAndConvert(trait.getConfigId()));
         // Invoke client creation
         // i.e.: Result<AtomicPrimitivesClient, Error> result = software.amazon.cryptography.primitives.internaldafny.__default.AtomicPrimitives(dafnyValue);
-        TypeName success = subject.dafnyNameResolver.classNameForConcreteServiceClient(subject.serviceShape);
+        TypeName success = subject.dafnyNameResolver.classNameForInterface(subject.serviceShape);
```

Comment:
Question/Issue: This is a large change, as the result type is now the abstract interface,
as compared to the concrete instance.

~~If we defined any methods on the concrete that are not included on the abstract,
we will be loosing access to them,
with out a cast.~~
~~Do you know why this change was necessary?~~

Having read the rest of the file, 
I see that we store it as an interface anyway.

Reply:
I saw that we had to update to the `index.dfy` files to return the interface.
This may bite us later, but we will resolve the issue then.